Subject:       FindTBB.cmake oneTBB support and increased verbosity
Patch author:  Orson Teodoro <orsonteodoro@hotmail.com>
Date:          Wed Jun 30 11:33:05 AM PDT 2021 (Unix time: 1625077985)

Copyright notice for nanovdb/cmake/FindTBB.cmake:

# Copyright Contributors to the OpenVDB Project
# SPDX-License-Identifier: MPL-2.0

diff -urp nanovdb.orig/cmake/FindTBB.cmake nanovdb/cmake/FindTBB.cmake
--- nanovdb.orig/cmake/FindTBB.cmake	2020-09-24 20:19:30.000000000 -0700
+++ nanovdb/cmake/FindTBB.cmake	2021-06-30 12:38:47.132155465 -0700
@@ -76,6 +76,9 @@ may be provided to tell this module wher
 
 #]=======================================================================]
 
+# oteodoro - increased verbosity of discovered versions and paths to observe correctness or bugs
+# oteodoro - add oneTTB checks
+
 cmake_minimum_required(VERSION 3.3)
 include(GNUInstallDirs)
 
@@ -90,7 +93,7 @@ mark_as_advanced(
 )
 
 set(_FIND_TBB_ADDITIONAL_OPTIONS "")
-if(DISABLE_CMAKE_SEARCH_PATHS)
+if(DISABLE_CMAKE_SEARCH_PATHS OR TBB_INCLUDEDIR)					# oteodoro - changed conditional
   set(_FIND_TBB_ADDITIONAL_OPTIONS NO_DEFAULT_PATH)
 endif()
 
@@ -148,14 +151,21 @@ list(APPEND _TBB_INCLUDE_SEARCH_DIRS
 )
 
 # Look for a standard tbb header file.
-find_path(Tbb_INCLUDE_DIR tbb/tbb_stddef.h
+find_path(Tbb_INCLUDE_DIR oneapi/tbb/version.h tbb/tbb_stddef.h				# oteodoro - changed line
   ${_FIND_TBB_ADDITIONAL_OPTIONS}
   PATHS ${_TBB_INCLUDE_SEARCH_DIRS}
   PATH_SUFFIXES ${CMAKE_INSTALL_INCLUDEDIR} include
 )
 
-if(EXISTS "${Tbb_INCLUDE_DIR}/tbb/tbb_stddef.h")
-  file(STRINGS "${Tbb_INCLUDE_DIR}/tbb/tbb_stddef.h"
+if(EXISTS ${Tbb_INCLUDE_DIR}/oneapi/tbb/version.h)					# oteodoro - added line
+  set(_tbb_version_file ${Tbb_INCLUDE_DIR}/oneapi/tbb/version.h)			# oteodoro - added line
+elseif(EXISTS "${Tbb_INCLUDE_DIR}/tbb/tbb_stddef.h")					# oteodoro - changed conditional instruction
+  set(_tbb_version_file "${Tbb_INCLUDE_DIR}/tbb/tbb_stddef.h")				# oteodoro - added line
+endif()											# oteodoro - added line
+message(STATUS "TBB include path: ${Tbb_INCLUDE_DIR}")					# oteodoro - added line
+
+if(EXISTS "${_tbb_version_file}")							# oteodoro - changed conditional
+  file(STRINGS "${_tbb_version_file}"							# oteodoro - changed path reference
     _tbb_version_major_string REGEX "#define TBB_VERSION_MAJOR "
   )
   string(REGEX REPLACE "#define TBB_VERSION_MAJOR" ""
@@ -163,7 +173,7 @@ if(EXISTS "${Tbb_INCLUDE_DIR}/tbb/tbb_st
   )
   string(STRIP "${_tbb_version_major_string}" Tbb_VERSION_MAJOR)
 
-  file(STRINGS "${Tbb_INCLUDE_DIR}/tbb/tbb_stddef.h"
+  file(STRINGS "${_tbb_version_file}"							# oteodoro - changed path reference
      _tbb_version_minor_string REGEX "#define TBB_VERSION_MINOR "
   )
   string(REGEX REPLACE "#define TBB_VERSION_MINOR" ""
@@ -175,6 +185,7 @@ if(EXISTS "${Tbb_INCLUDE_DIR}/tbb/tbb_st
   unset(_tbb_version_minor_string)
 
   set(Tbb_VERSION ${Tbb_VERSION_MAJOR}.${Tbb_VERSION_MINOR})
+  message(STATUS "TBB version:  ${Tbb_VERSION}")					# oteodoro - added line
 endif()
 
 # ------------------------------------------------------------------------
@@ -239,6 +250,7 @@ foreach(COMPONENT ${TBB_FIND_COMPONENTS}
   else()
     set(TBB_${COMPONENT}_FOUND FALSE)
   endif()
+  message(STATUS "TBB ${COMPONENT} lib path: ${Tbb_${COMPONENT}_LIBRARY}")				# oteodoro - added line
 endforeach()
 
 # Reset library suffix
